ti/instrumentation/utils.go

```diff
@@ -31,8 +33,114 @@ const (
 	gitBin       = "git"
 	outDir       = "%s/ti/callgraph/" // path passed as outDir in the config.ini file
 	tiConfigPath = ".ticonfig.yaml"
+	// Parallelism environment variables
+	harnessStepIndex  = "HARNESS_STEP_INDEX"
+	harnessStepTotal  = "HARNESS_STEP_TOTAL"
+	harnessStageIndex = "HARNESS_STAGE_INDEX"
+	harnessStageTotal = "HARNESS_STAGE_TOTAL"
 )
 
+func getTestTime(ctx context.Context, splitStrategy string) (map[string]float64, error) {
+	fileTimesMap := map[string]float64{}
+	// Get the timing data
+	cfg := pipeline.GetState().GetTIConfig()
+	if cfg == nil || cfg.URL == "" {
+		return fileTimesMap, fmt.Errorf("TI config is not provided in setup")
+	}
+	c := client.NewHTTPClient(cfg.URL, cfg.Token, cfg.AccountID, cfg.OrgID, cfg.ProjectID,
+		cfg.PipelineID, cfg.BuildID, cfg.StageID, cfg.Repo, cfg.Sha, false)
+
+	req := ti.GetTestTimesReq{}
+	var res ti.GetTestTimesResp
+	var err error
+
+	switch splitStrategy {
+	case testsplitter.SplitByFileTimeStr:
+		req.IncludeFilename = true
+		res, err = c.GetTestTimes(ctx, &req)
+		fileTimesMap = testsplitter.ConvertMap(res.FileTimeMap)
+	case testsplitter.SplitByClassTimeStr:
+		req.IncludeClassname = true
+		res, err = c.GetTestTimes(ctx, &req)
+		fileTimesMap = testsplitter.ConvertMap(res.ClassTimeMap)
+	case testsplitter.SplitByTestcaseTimeStr:
+		req.IncludeTestCase = true
+		res, err = c.GetTestTimes(ctx, &req)
+		fileTimesMap = testsplitter.ConvertMap(res.TestTimeMap)
+	case testsplitter.SplitByTestSuiteTimeStr:
+		req.IncludeTestSuite = true
+		res, err = c.GetTestTimes(ctx, &req)
+		fileTimesMap = testsplitter.ConvertMap(res.SuiteTimeMap)
+	case testsplitter.SplitByFileSizeStr:
+		return map[string]float64{}, nil
+	default:
+		return map[string]float64{}, nil
+	}
+	if err != nil {
+		return map[string]float64{}, err
+	}
+	return fileTimesMap, nil
+}
+
+// getSplitTests takes a list of tests as input and returns the slice of tests to run depending on
+// the test split strategy and index
+func getSplitTests(ctx context.Context, log *logrus.Logger, testsToSplit []ti.RunnableTest, splitStrategy string, splitIdx, splitTotal int) ([]ti.RunnableTest, error) {
+	if len(testsToSplit) == 0 {
+		return testsToSplit, nil
+	}
+
+	currentTestMap := make(map[string][]ti.RunnableTest)
+	currentTestSet := make(map[string]bool)
+	var testID string
+	for _, t := range testsToSplit {
+		switch splitStrategy {
+		case classTimingTestSplitStrategy, countTestSplitStrategy:
+			testID = t.Pkg + t.Class
+		default:
```

Comment:
same testId as prev case

Reply:
Added switch case for future proofing - when we support more strategies.